### PR TITLE
Add Unicode regex tests

### PR DESCRIPTION
**Pull request by f3ath**
I realized that my implementation did not support I-Regex, so adding some tests for Unicode in regex.

### CODE REVIEW

**Comment by glyn** [comment]
I'm afraid I haven't paid much attention to i-regexp, so I can't really comment on the details of this PR, but it does raise the interesting question of how much i-regexp function it would be appropriate to test in this CTS.

**Comment by f3ath** [comment]
IMO the CTS should test that the implementation uses IRegexp and not a whatever built-in implementation. Adding a comprehensive IRegexp specific test would probably be an overkill, but we can test for some notable IRegexp features. I propose the checks list:
- Basic Unicode compatibility (that's what this PR is targeting)
- The `.` matcher behavior (matches `\u2028` and `\u2029`)
- Does NOT support groups, anchoring, char classes like `\d` and `\p{IsBasicLatin}`)

**Comment by gregsdennis** [comment]
@f3ath you weren't around then, but the entire purpose of creating `iregexp` was to serve as a minimum interface that most regex providers would already support.  It's actually _expected_ that most Path implementations would use what's at their disposal for regex support.  Because of that, I think it's counterproductive to have negative tests around regex.

**Comment by f3ath** [comment]
Fair enough, but I'm still confused about the desired dot matcher behavior. I build JSONPath implementation in Dart which uses ecmascript regexps. As pointed out in the IRegexp spec, the dot match is slightly different between IRegexp and ecmascript. Does the JSONPath spec care about such nuances or is it totally up to the implementers? Do we at least want a test for Unicode support in regexps, like the one we have for length()?

**Comment by glyn** [comment]
> Fair enough, but I'm still confused about the desired dot matcher behavior. I maintain JSONPath implementation in Dart which uses ecmascript regexps. As pointed out in the IRegexp spec, the dot match is slightly different between IRegexp and ecmascript. Does the JSONPath spec care about such nuances or is it totally up to the implementers? Do we at least want a test for Unicode support in regexps, like the one we have for length()?

JSONPath needs to be interoperable and therefore so does i-regexp.

@cabo may be interested in this topic.

**Comment by cabo** [comment]
(writing from memory)

```
match("🤔", "[🤔]")
match("\u2028", ".")
```

should be good tests for non-BMP characters (unicode property in ECMAscript) and for handling dot in the presence of line separators.

File: cts.json [hunk 1]
```diff
@@ -2429,6 +2446,24 @@
...
+      "selector": "$[?search(@, '\\\\P{Lu}')]",
+      "document": ["ж", "Ж", "1", true, [], {}],
+      "result": ["ж", "1"]
+    },
+    {
+      "name": "filter, search function, unicode, surrogate pair",
+      "selector": "$[?search(@, 'a.b')]",
+      "document": ["a\uD800\uDD01bc", "abc", "1", true, [], {}],
+      "result": ["a\uD800\uDD01bc"]
+    },
```
[the remark this answers is not in the document]
Comment (gregsdennis) [reply cts.json:2466]: Such a test should be provided in so we know which implementations do not support non-BMP Unicode.

This should be added to the comparison site, then, not the test suite.

[the remark this answers is not in the document]
Comment (cabo) [reply cts.json:2466]: This deviation is about implementation bugs, not about features.

(I could also imagine that ECMAScript implementations will ignore the LS/PS problem mentioned in Section 5.3 of iregexp.)

[the remark this answers is not in the document]
Comment (gregsdennis) [reply cts.json:2466]: This may just be a .Net issue.  I may have to ignore this test, then.

Ref: https://github.com/ietf-wg-jsonpath/iregexp/issues/22#issuecomment-1510520470

**Review by gregsdennis — APPROVED** [review]

**Comment by gregsdennis** [comment]
Needs a rebase since the test file split PR was merged.

File: cts.json [hunk 2]
```diff
@@ -1750,7 +1750,7 @@
       ]
     },
     {
-      "name": "index selector, index selector",
```
Comment (f3ath) [inline cts.json:1753]: I guess I forgot to run the build command last time, hence these artifacts.

**Comment by gregsdennis** [comment]
I'm going to go ahead and merge this.  If there is something that needs changing, let's create a follow-up PR.
